plugins/scaffolder-backend-module-gitlab/src/actions/createGitlabIssueAction.ts

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InputError } from '@backstage/errors';
+import { ScmIntegrationRegistry } from '@backstage/integration';
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import commonGitlabConfig from '../commonGitlabConfig';
+import { examples } from './createGitlabIssueAction.examples';
+import { z } from 'zod';
+import { checkEpicScope, convertDate, getClient, parseRepoUrl } from '../util';
+import { Gitlab, CreateIssueOptions, IssueSchema } from '@gitbeaker/rest';
+
+const enumIssueType = {
+  ISSUE: 'issue',
+  INCIDENT: 'incident',
+  TEST: 'test_case',
+};
+
+const issueInputProperties = z.object({
+  projectId: z.number().describe('Project Id'),
+  title: z.string({ description: 'Title of the issue' }),
+  assignees: z
+    .array(z.number(), {
+      description: 'IDs of the users to assign the issue to.',
+    })
+    .optional(),
+  confidential: z.boolean({ description: 'Issue Confidentiality' }).optional(),
+  description: z.string().describe('Issue description').max(1048576).optional(),
+  createdAt: z
+    .string()
+    .describe('Creation date/time')
+    .regex(
+      /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d{3})?Z$/,
+      'Invalid date format. Use YYYY-MM-DDTHH:mm:ssZ or YYYY-MM-DDTHH:mm:ss.SSSZ',
+    )
+    .optional(),
+  dueDate: z
+    .string()
+    .describe('Due date/time')
+    .regex(
+      /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d{3})?Z$/,
+      'Invalid date format. Use YYYY-MM-DDTHH:mm:ssZ or YYYY-MM-DDTHH:mm:ss.SSSZ',
+    )
+    .optional(),
+  discussionToResolve: z
+    .string({
+      description:
+        'Id of a discussion to resolve. Use in combination with "merge_request_to_resolve_discussions_of"',
+    })
+    .optional(),
+  epicId: z
+    .number({ description: 'Id of the linked Epic' })
+    .min(0, 'Valid values should be equal or greater than zero')
+    .optional(),
+  labels: z.string({ description: 'Labels to apply' }).optional(),
+  issueType: z
+    .string({
```

Comment:
I wonder if it's possible to use `nativeEnums` here instead?

https://github.com/colinhacks/zod#native-enums

Reply:
Hi guys,

it's done. Please let me know what you think! :)